packages/atlas/src/views/viewer/MemberView/MemberActivity.hooks.ts

```diff
@@ -0,0 +1,193 @@
+import { useMemo } from 'react'
+
+import { useRawActivities } from '@/api/hooks'
+import { BasicMembershipFieldsFragment, StorageDataObjectFieldsFragment } from '@/api/queries'
+
+export type NftActivitiesRecord = {
+  id?: string
+  date?: Date
+  block?: number
+  video?: {
+    id: string
+    title: string
+    thumbnailPhoto: StorageDataObjectFieldsFragment | null
+  }
+}
+export type ActivitiesRecord =
+  | ({
+      type: 'Bid'
+      bidAmount: number
+      from: BasicMembershipFieldsFragment
+    } & NftActivitiesRecord)
+  | ({
+      type: 'Purchase'
+      from: BasicMembershipFieldsFragment | null
+      price: number
+      to: BasicMembershipFieldsFragment | null
+    } & NftActivitiesRecord)
+  | ({
+      type: 'Withdrawal'
+      from: BasicMembershipFieldsFragment
+    } & NftActivitiesRecord)
+  | ({
+      type: 'Sale'
+      from: BasicMembershipFieldsFragment | null
+      price: number
+      to: BasicMembershipFieldsFragment | null
+    } & NftActivitiesRecord)
+  | ({
+      type: 'Removal'
+      from: BasicMembershipFieldsFragment | null
+    } & NftActivitiesRecord)
+  | ({
+      type: 'Transfer'
+      from: BasicMembershipFieldsFragment
+    } & NftActivitiesRecord)
+  | ({
+      type: 'Listing'
+      typeName: 'EnglishAuctionStartedEvent' | 'OpenAuctionStartedEvent' | 'NftSellOrderMadeEvent'
+      from: BasicMembershipFieldsFragment | null
+      price?: number
+    } & NftActivitiesRecord)
+  | ({
+      type: 'Mint'
+      from: BasicMembershipFieldsFragment | null
+    } & NftActivitiesRecord)
+  | ({
+      type: 'Price change'
+      from: BasicMembershipFieldsFragment | null
+      price: number
+    } & NftActivitiesRecord)
+
+const parseActivities = (
+  event: ReturnType<typeof useRawActivities>['activities'][0],
```

Comment:
We don't need to grab the first item of the array we can just do: 
```suggestion
  event: ReturnType<typeof useRawActivities>['activities'][number],
```